core/inspector.rs

```diff
@@ -236,20 +259,163 @@ impl JsRuntimeInspector {
+          });
 
+          break;
+        }
+        PollState::BlockOnWaitingForSession => {
+          futures::executor::block_on(futures::future::poll_fn(|cx| {
+            // Accept new connections.
+            let poll_result = sessions.session_rx.poll_next_unpin(cx);
+            if let Poll::Ready(Some(session_proxy)) = poll_result {
+              let mut session = InspectorSession::new(
+                self.v8_inspector.clone(),
+                session_proxy,
+                false,
+              );
+
+              if let Poll::Ready(Some(session_stream_item)) =
+                session.poll_next_unpin(cx)
+              {
+                let (v8_session_ptr, msg) = session_stream_item;
+                InspectorSession::dispatch_message(v8_session_ptr, msg);
+              }
+              sessions.established.push(session);
+            }
+
+            loop {
+              let poll_result = sessions.established.poll_next_unpin(cx);
+              match poll_result {
+                Poll::Ready(Some(session_stream_item)) => {
+                  let (v8_session_ptr, msg) = session_stream_item;
+                  InspectorSession::dispatch_message(v8_session_ptr, msg);
+                }
+                _ => break,
+              }
+            }
+
+            // Keep polling until a session connects and tell us to start
+            // running.
+            if self.flags.borrow().waiting_for_session {
+              Poll::Pending
+            } else {
+              Poll::Ready(())
+            }
+          }));
+          self.waker.update(|w| {
+            assert_eq!(w.poll_state, PollState::BlockOnWaitingForSession);
+            w.poll_state = PollState::SyncPolling;
+          });
+        }
+        PollState::BlockOnPause => {
+          futures::executor::block_on(futures::future::poll_fn(|cx| {
+            // Accept new connections.
+            let poll_result = sessions.session_rx.poll_next_unpin(cx);
+            if let Poll::Ready(Some(session_proxy)) = poll_result {
+              let mut session = InspectorSession::new(
+                self.v8_inspector.clone(),
+                session_proxy,
+                false,
+              );
+
+              if let Poll::Ready(Some(session_stream_item)) =
+                session.poll_next_unpin(cx)
+              {
+                let (v8_session_ptr, msg) = session_stream_item;
+                InspectorSession::dispatch_message(v8_session_ptr, msg);
+              }
+              sessions.established.push(session);
+            }
+
+            // Keep polling until a resume execution from a pause.
+            let poll_result = sessions.established.poll_next_unpin(cx);
+            match poll_result {
+              Poll::Ready(Some(session_stream_item)) => {
+                let (v8_session_ptr, msg) = session_stream_item;
+                InspectorSession::dispatch_message(v8_session_ptr, msg);
+                Poll::Ready(())
```

Comment:
In the comment 6 lines up it says "// Keep polling until a resume execution from a pause.".

However that's not what happens here. You poll until there is one message to dispatch, and then you call it a day.